### PR TITLE
Initial manifest for org.stellarium.Stellarium

### PR DESCRIPTION
**Pull request by cosimoc**

### CODE REVIEW

File: org.stellarium.Stellarium.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+    "app-id": "org.stellarium.Stellarium",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.10",
```
Comment (TingPing) [inline org.stellarium.Stellarium.json]: Choose 5.9 (LTS) or 5.11.

File: org.stellarium.Stellarium.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/man",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name": "stellarium",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline org.stellarium.Stellarium.json]: `cmake-ninja`

File: org.stellarium.Stellarium.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+    "app-id": "org.stellarium.Stellarium",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.10",
+    "sdk": "org.kde.Sdk",
+    "command": "stellarium",
+    "rename-icon": "stellarium",
+    "rename-appdata-file": "stellarium.appdata.xml",
```
Comment (TingPing) [inline org.stellarium.Stellarium.json]: Upstreams file is missing `releases`

**Comment by TingPing** [comment]
Also just to be frank, you need to be a more active maintainer or you need upstream involved. Many of your packages here are just dumped once uploaded but this application is actively developed upstream and is using a relatively quick moving runtime and needs somebody more hands on.

**Comment by cosimoc** [comment]
@TingPing Thanks for the feedback... Yeah, I am conscious that I haven't been able to catch up much with application maintenance for Flathub lately. I will try my best to stay more involved or be more proactive about involving upstream from now on.

@alex-w I created a flatpak build manifest for Stellarium to be published on Flathub. Since on Flathub we encourage upstreams to maintain their applications independently, are you interested in contributing to keeping it up to date in the future?

**Comment by cosimoc** [comment]
@TingPing I filed an upstream PR for the missing appdata tags, and backported it here. Also addressed the other review comments.

**Comment by TingPing** [comment]
bot, build org.stellarium.Stellarium

File: stellarium-improve-appdata.patch [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
++    <release date="2018-07-01" version="0.18.1"/>
++  </releases>
+   <screenshots>
+     <screenshot type="default">
+       <image>https://www.stellarium.org/img/screenshots/0.10-constellation-art-on.jpg</image>
+@@ -418,6 +421,35 @@
+       <image>https://stellarium.org/img/screenshots/0.10-orion-nebula.jpg</image>
+     </screenshot>
+   </screenshots>
++  <content_rating type="oars-1.1">
```
Comment (nedrichards) [inline stellarium-improve-appdata.patch]: Not a blocker, but there's a new compact notation you can use now:
>   <content_rating type="oars-1.1" />
That's the same as all none.

**Comment by cosimoc** [comment]
Updated to use the compact syntax.
Also FWIW the previous build seems to have worked fine: https://flathub.org/builds/#/builders/2/builds/3108

**Comment by alex-w** [comment]
@cosimoc Sorry, I'm not familiar to flatpak packages format and it looks very weird for me. For example I do not understand why flatpak for Stellarium requires the huge runtime (around 1.1GB of size). Stellarium want Qt, but not all KDE! 

As I said before it looks very weird for me and I'm do not have free time to investigate specific stuff for flatpak, so, sorry, but I'm do not want to support this packages. But, of course, we welcoming the patches into upstream.

**Comment by cosimoc** [comment]
@alex-w Thanks for the feedback. I used the KDE runtime because it already includes a full copy of QT, and since many other applications use it, it's very likely users will already have it installed on their system. It would be possible to make it use the Freedesktop runtime and bundle a separate copy of QT, but this is a lot more practical and the KDE team already maintains QT in their runtime.
See https://blogs.gnome.org/mclasen/2018/06/13/flatpak-in-detail/ for a good overview of how it all fits together.

At the end of the day disk space is kind of cheap, but note that Flatpak uses ostree in the backend to deduplicate installed files on disk, so there are real space savings when multiple runtimes and applications are installed. See http://docs.flatpak.org/en/latest/under-the-hood.html for a technical background if you're interested!

Either way, I hope you can consider this in the future as a direct way for your software to reach users. In the meantime I'd be happy to keep maintaining this as my time allows.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.stellarium.Stellarium

**Pull request closed without merging**